python/cudf/cudf/core/indexing.py

```diff
@@ -164,17 +166,23 @@ def __setitem__(self, key, value):
         self._sr.iloc[key] = value
 
     def _loc_to_iloc(self, arg):
-        from cudf.core.column import column
-        from cudf.core.series import Series
-
         if is_scalar(arg):
-            try:
-                found_index = self._sr.index._values.find_first_value(
-                    arg, closest=False
-                )
+            if isinstance(arg, int):
```

Comment:
And won't work if arg is `np.int`...

```
>>> s = pd.Series([1 ,2 ,3], index=["a", "b", "c"])
>>> s[np.int32(1)]
2
>>> s[np.int8(0)]
1
```

Reply:
All of these return 2 though, please elaborate? 
```
In [10]: ps
Out[10]: 
a    1
b    2
c    3
dtype: int64

In [11]: ps[np.int8(1)]
Out[11]: 2

In [12]: ps[1]
Out[12]: 2

In [13]: ps[np.int32(1)]
Out[13]: 2

```

Reply:
I found a function that might be useful to check for valid integer objects: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.api.types.is_integer.html#

In pandas `int`, and `np.integer` are ints, and bools are not ints. In python, `Bool` is `Int`.
```
>>> pd.api.types.is_integer(100)
True
>>> pd.api.types.is_integer(-100)
True
>>> pd.api.types.is_integer(np.uint16(10))
True
>>> pd.api.types.is_integer(np.int8(-42))
True
>>> pd.api.types.is_integer(True)
False
```